pennylane/devices/default_clifford.py

```diff
@@ -0,0 +1,644 @@
+                        results=res,
+                        shots=shots,
+                        resources=c.specs["resources"],
+                    )
+                else:
+                    self.tracker.update(
+                        simulations=1,
+                        executions=qpu_executions,
+                        results=res,
+                        resources=c.specs["resources"],
+                    )
+                self.tracker.record()
+
+        return results[0] if is_single_circuit else results
+
+    def supports_derivatives(
+        self,
+        execution_config: Optional[ExecutionConfig] = None,
+        circuit: Optional[QuantumTape] = None,
+    ) -> bool:
+        """Check whether or not derivatives are available for a given configuration and circuit.
+
+        ``DefaultClifford`` returns trivial derivates everytime.
+
+        Args:
+            execution_config (ExecutionConfig): The configuration of the desired derivative calculation
+            circuit (QuantumTape): An optional circuit to check derivatives support for.
+
+        Returns:
+            Bool: Whether or not a derivative can be calculated provided the given information
+
+        """
+
+        return True
+
+    def compute_derivatives(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        is_single_circuit = False
+        if isinstance(circuits, QuantumScript):
+            is_single_circuit = True
+            circuits = [circuits]
+
+        if self.tracker.active:
+            self.tracker.update(derivative_batches=1, derivatives=len(circuits))
+            self.tracker.record()
+
+        res = tuple(tuple(np.zeros(s) for s in circuit.shape(self)) for circuit in circuits)
+
+        return res[0] if is_single_circuit else res
+
+    # pylint:disable=no-member
+    def simulate(
+        self,
+        circuit: qml.tape.QuantumScript,
+        debugger=None,
+    ) -> Result:
+        """Simulate a single quantum script.
+
+        Args:
+            circuit (QuantumTape): The single circuit to simulate
+            debugger (_Debugger): The debugger to use
+
+        Returns:
+            tuple(TensorLike): The results of the simulation
+
+        This function assumes that all operations are Clifford.
+
+        >>> qs = qml.tape.QuantumScript([qml.Hadamard(wires=0)], [qml.expval(qml.PauliZ(0)), qml.state()])
+        >>> qml.devices.DefaultClifford().simulate(qs)
+        (array(0),
+         array([[0, 1, 0],
+                [1, 0, 0]]))
+
+        """
+
+        stim = _import_stim()
+
+        # Account for custom labelled wires
+        # TODO: Fix custom integer ordering :)
+        circuit = circuit.map_to_standard_wires()
+
+        if circuit.shots:
+            raise NotImplementedError(
+                "default.clifford currently doesn't support computation with shots."
+            )
```

Comment:
When do we plan to add finite-shot support? How much work would be needed?

Reply:
Finite-shot support is ready for review in this PR - https://github.com/PennyLaneAI/pennylane/pull/4954. ✨